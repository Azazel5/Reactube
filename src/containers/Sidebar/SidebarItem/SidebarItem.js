import React from 'react'
import './SidebarItem.scss'
import { Menu, Icon } from 'semantic-ui-react'

const sidebarItem = (props) => {
    const highlight = props.highlight ? 'highlight-item' : null
    return (
        <Menu.Item className={['sidebar-item', highlight].join(' ')}>
            <div className="sidebar-item-alignment-container">
                <span><Icon size="large" name={props.icon} /></span>
                <span>{props.label}</span>
            </div>
        </Menu.Item>
    )
}

export default sidebarItem