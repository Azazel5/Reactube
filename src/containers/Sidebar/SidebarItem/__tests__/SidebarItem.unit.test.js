import React from 'react'
import {shallow} from 'enzyme'
import SidebarItem from '../SidebarItem'

describe('<SidebarItem />', () => {
    test('renders empty <SidebarItem /> item', () => {
        const wrapper = shallow(<SidebarItem />)
        expect(wrapper).toMatchSnapshot()
    })

    test('renders <SidebarItem /> with non-highlighted props', () => {
        const wrapper = shallow(<SidebarItem label='Trending' icon='fire'/>)
        expect(wrapper).toMatchSnapshot()
    })

    test('renders <SidebarItem /> with highlighted props', () => {
        const wrapper = shallow(<SidebarItem highlighted label='Trending' icon='fire'/>)
        expect(wrapper).toMatchSnapshot()
    })
})