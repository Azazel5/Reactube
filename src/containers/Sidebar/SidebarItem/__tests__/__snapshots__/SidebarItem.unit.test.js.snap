// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarItem /> renders <SidebarItem /> with highlighted props 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`<SidebarItem /> renders <SidebarItem /> with non-highlighted props 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`<SidebarItem /> renders empty <SidebarItem /> item 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;
